repository/gridftp/globus5/authz/tests/unit/test_AuthzWorker.cpp

```diff
@@ -0,0 +1,409 @@
+  SDMS::AuthzWorker worker(&config, log_context);
+  char trailing_slash_url[] = "ftp://hostname/path/";
+  std::string expected_result = "/path/";
+  BOOST_CHECK_EQUAL(worker.removeOrigin(trailing_slash_url), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(RemoveOriginRootPathTest) {
+  // Test an FTP URL pointing to the root path
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char root_path_url[] = "ftp://hostname/";
+  std::string expected_result = "/";
+  BOOST_CHECK_EQUAL(worker.removeOrigin(root_path_url), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(RemoveOriginDeepPathTest) {
+  // Test an FTP URL with a deep path
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char deep_path_url[] = "ftp://hostname/a/b/c/d/e.txt";
+  std::string expected_result = "/a/b/c/d/e.txt";
+  BOOST_CHECK_EQUAL(worker.removeOrigin(deep_path_url), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(GetAuthzPathValidPathTest) {
+  // Test a valid full FTP path
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char valid_path[] = "ftp://hostname/globus/root/path/to/resource.txt";
+  std::string expected_result = "/path/to/resource.txt";
+
+  BOOST_CHECK_EQUAL(worker.getAuthzPath(valid_path), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(GetAuthzPathInvalidFTPPathTest) {
+  // Test an invalid FTP path (removeOrigin will fail)
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char invalid_ftp_path[] = "http://hostname/globus/root/path/to/resource.txt";
+
+  BOOST_CHECK_THROW(worker.getAuthzPath(invalid_ftp_path), TraceException);
+}
+
+BOOST_AUTO_TEST_CASE(GetAuthzPathInvalidPOSIXPathTest) {
+  // Test an invalid POSIX path (isPathValid will fail)
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char invalid_posix_path[] = "ftp://hostname/wrong_root/path/to/resource.txt";
+
+  BOOST_CHECK_THROW(worker.getAuthzPath(invalid_posix_path), TraceException);
+}
+
+BOOST_AUTO_TEST_CASE(GetAuthzPathRootPathTest) {
+  // Test a valid full FTP path pointing to the root
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char root_path[] = "ftp://hostname/globus/root/";
+  std::string expected_result = "/";
+
+  BOOST_CHECK_EQUAL(worker.getAuthzPath(root_path), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(GetAuthzPathDeepPathTest) {
+  // Test a valid full FTP path with a deep path
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  char deep_path[] = "ftp://hostname/globus/root/a/b/c/d/e.txt";
+  std::string expected_result = "/a/b/c/d/e.txt";
+
+  BOOST_CHECK_EQUAL(worker.getAuthzPath(deep_path), expected_result);
+}
+
+BOOST_AUTO_TEST_CASE(ProcessResponseWithTimeout) {
+  SDMS::LogContext log_context;
+  SDMS::AuthzWorker worker(&config, log_context);
+  SDMS::ICommunicator::Response response;
+  response.time_out = true;
```

Comment:
I'm curious how this timeout works in practice. Is this processing job added to a queue?